### PR TITLE
feat(instance) Introduce `Instance.call_function_by_index`

### PR DESCRIPTION
**Pull request by Hywan**
It depends on https://github.com/wasmerio/wasmer/pull/1306.

As is, it can be merged with my current fork of Wasmer. cc @syrusakbary, thoughts?

### CODE REVIEW

File: src/instance/mod.rs [hunk 1]
```diff
@@ -141,4 +165,56 @@ impl Instance {
...
+                        .exports()
+                        .filter(|(_, export)| match export {
+                            Export::Function { .. } => true,
+                            _ => false,
+                        })
+                        .map(|(name, _)| (self.instance.resolve_func(&name).unwrap(), name.clone()))
+                        .collect(),
+                );
+
+                dbg!(&self.exports_index_to_name);
```
Comment (Hywan) [inline src/instance/mod.rs]: Oops…

File: src/instance/exports.rs [hunk 2]
```diff
@@ -85,6 +85,78 @@ impl InspectExportedFunction for ExportedFunction {
...
+) -> PyResult<PyObject> {
+    // Check the given arguments match the exported function signature.
+    let signature = function.signature();
+    let parameters = signature.params();
+
+    let number_of_parameters = parameters.len() as isize;
+    let number_of_arguments = arguments.len() as isize;
+    let diff: isize = number_of_parameters - number_of_arguments;
+
+    match diff.cmp(&0) {
```
Comment (syrusakbary) [inline src/instance/exports.rs]: Why a match instead of if/else ? (that might be more readable)

Comment (Hywan) [reply src/instance/exports.rs]: It makes Clippy happy.

**Pull request merged**